src/components/application_manager/src/application_impl.cc

```diff
@@ -442,12 +443,18 @@ void ApplicationImpl::StopStreaming(
 
   SuspendStreaming(service_type);
 
-  if (service_type == ServiceType::kMobileNav && video_streaming_approved()) {
-    StopNaviStreaming();
-  } else if (service_type == ServiceType::kAudio &&
-             audio_streaming_approved()) {
+  if (!helpers::Compare<ServiceType, helpers::EQ, helpers::ONE>(
+          service_type, ServiceType::kMobileNav, ServiceType::kAudio)) {
+    return;
+  }
+
+  if (audio_streaming_approved()) {
     StopAudioStreaming();
   }
+
+  if (video_streaming_approved()) {
+    StopNaviStreaming();
+  }
```

Comment:
Because there is only one service type parameter being passed to this function I think it would be cleaner to only stop the service passed to the stop streaming function, instead of trying to stop both audio and video services in the same function.

Reply:
@JackLivio Agree. It would be cleaner. But in case we exit all applications - we have to stop all types of streamings approved.